classes/notification/NotificationManager.inc.php

```diff
@@ -158,6 +158,19 @@ protected function getMgrDelegate($notificationType, $assocType, $assocId) {
 		return parent::getMgrDelegate($notificationType, $assocType, $assocId);
 	}
 
+	/**
+	 * @copydoc PKPNotificationManager::getNotificationSettingsMap()
+	 */
+	public function getNotificationSettingsMap() {
```

Comment:
This overrides a protected function, which will cause a warning/error, I think.

Reply:
Although it does not produce a warning it definitely needs addressing. (I think the reason it does not complain is that it thinks I overload the function rather than overriding it, and then, inside, I am using the parent function (which is OK). 

The exposed function is, in fact the child's `getNotificationSettingsMap` function. So here we have 2 solutions I think (except the one already implemented).
1. Change the visibility of parent's `getNotificationSettingsMap` to public.
2. Implementing a "getter" for the SettingsMap at the child class level which will expose the array, calling only the `getNotificationSettingsMap` inside, and changing the `getNotificationSettingsMap` to `protected`. I will go for this one.

Reply:
Agreed, I think making the parent class implementation `public` is the best option!